### PR TITLE
#1 patch of fixes.

### PR DESCRIPTION
**Pull request by anatolydyachenko**
Project with latest changes was found. Also I fixed some issues from comments

### CODE REVIEW

File: DogServiceExample/pom.xml [hunk 1]
```diff
@@ -2,11 +2,10 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.common</groupId>
-  <artifactId>DogExample</artifactId>
+  <artifactId>Dog-Example</artifactId>
```
Comment (korobochka) [inline DogServiceExample/pom.xml]: Small letters => `dog-example`

File: DogServiceExample/pom.xml [hunk 2]
```diff
@@ -51,29 +50,13 @@
...
-    </dependency>
-    <dependency>
       <groupId>com.google.code.gson</groupId>
       <artifactId>gson</artifactId>
       <version>2.2.4</version>
     </dependency>
+
     <dependency>
       <groupId>com.googlecode.json-simple</groupId>
       <artifactId>json-simple</artifactId>
```
Comment (korobochka) [inline DogServiceExample/pom.xml]: still both gson and json-simple are present, choose one

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    private int height;
+    private int weight;
+
+    public Dog() {
+        id = -1;
+        name = null;
+        dateOfBirth = null;
+        height = -1;
+        weight = -1;
+        DogService.incrementDogId();
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: This introduces two way dependency, which makes application more comples.
Ideally your Dog should know nothing about DogService.

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        if (name.length() >= 1 && name.length() <= 100) {
+            this.name = name;
+        } else {
+            System.out.println("Name should has length in range 1-100. Value is not updated to next one: " + name);
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: exceptions

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    public void setDateOfBirth(String dateOfBirth) {
+        DateFormat simpleDateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        try {
+            Date currentDate = simpleDateFormat.parse(simpleDateFormat.format(new Date()));
+            Date dateOfBirthDate = simpleDateFormat.parse(dateOfBirth);
+
+            if (dateOfBirthDate.before(currentDate))
+                this.dateOfBirth = dateOfBirth;
+            else {
+                System.out.println("DateOfBirth should be before current date. Value is not updated. DateOfBirth date: " + dateOfBirth + ". Current date: " + simpleDateFormat.format(new Date()));
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: exceptions

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+    }
+
+    public String getDateOfBirth() {
+        return dateOfBirth;
+    }
+
+    public void setDateOfBirth(String dateOfBirth) {
+        DateFormat simpleDateFormat = new SimpleDateFormat("dd.MM.yyyy");
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: extract constant into static final class field

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        }
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void setHeight(int height) {
+        if (height > 0)
+            this.height = height;
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: braces around branches

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        else
+            System.out.println("Height should be positive: " + height + ". Value is not updated");
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: exception

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public void setWeight(int weight) {
+        if (weight > 0)
+            this.weight = weight;
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: braces

File: DogServiceExample/src/main/java/com/common/app/Dog.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        else
+            System.out.println("Weight should be positive: " + weight + ". Value is not updated");
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/app/Dog.java]: exceptions

File: DogServiceExample/src/main/java/com/common/json/JsonHelper.java [hunk 11]
```diff
@@ -15,4 +16,10 @@ public static String mapToJson(Map<Integer, Dog> map) {
             map = new HashMap<Integer, Dog>();
         return gson.toJson(map);
     }
+
+    public static String JsonObjectToJson(JsonObject jsonObject) {
+        if (jsonObject == null)
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/json/JsonHelper.java]: Are this null-checks important? It seems to me that if you pass `null` here it is a logical error in your program, should handle it appropriately in some other place

File: DogServiceExample/src/main/java/com/common/rest/DogService.java [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.gson.JsonParser;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Path("/")
+public class DogService {
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/rest/DogService.java:17]: Here you convert method parameters from Json and responses to Json manually. There mush be a way to do this automatically: you can define function like `public Dog createNewDog(Dog dog)` and the parameters and responses will be converted by the application server.

Comment (korobochka) [reply DogServiceExample/src/main/java/com/common/rest/DogService.java:17]: This way you will not have to set status 200 manually, for example. And if you throw exceptions in Dog setters (if not passed validation) status will be set to 500 automatically.

File: DogServiceExample/src/main/java/com/common/rest/DogService.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+        return Response.status(200).entity(JsonHelper.mapToJson(dogs)).build();
+    }
+
+    @GET
+    @Path("dog/{id}")
+    @Produces("application/json")
+    public Response printDog(@PathParam("id") String id) {
+        JsonElement jelement = new JsonParser().parse(JsonHelper.mapToJson(dogs));
+        JsonObject jobject = jelement.getAsJsonObject();
+        JsonObject result = jobject.getAsJsonObject(id);
```
Comment (korobochka) [inline DogServiceExample/src/main/java/com/common/rest/DogService.java]: Ssame comments as in previous review, just get the dog from the map.
And don't forget to return 404 if no dog with such Id.

File: DogServiceExample/src/test/java/com/common/war/DogTest.java [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+        dog1.setWeight(0);
+        Assert.assertEquals(dog1.getWeight(), tmpWeight);
+
+        dog1.setWeight(1);
+        Assert.assertEquals(dog1.getWeight(), 1);
+    }
+
+    @Test
+    public void getForDog() {
+        Dog dog1 = new Dog(10, "TestName", "25.05.2010", 15, 25, true);
```
Comment (korobochka) [inline DogServiceExample/src/test/java/com/common/war/DogTest.java]: Usually it does not make much sense to test getters.

File: DogServiceExample/src/test/java/com/common/war/DogTest.java [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+        dog1.setName("q"); //try to enter min value
+        Assert.assertEquals(dog1.getName(), "q");
+
+        dog1.setName("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"); //try to enter max value
+        Assert.assertEquals(dog1.getName(), "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890");
+    }
+
+    @Test
+    public void setDateOfBirthForDog() {
```
Comment (korobochka) [inline DogServiceExample/src/test/java/com/common/war/DogTest.java]: same comments about splitting into multiple tests

**Review by korobochka — CHANGES_REQUESTED** [review]
Review round 2.
You can push additional commits to the same branch `master_update` to fix the issues.

File: DogServiceExample/pom.xml (hunk 1 above)
Comment (anatolydyachenko) [reply DogServiceExample/pom.xml]: changed

File: DogServiceExample/pom.xml (hunk 2 above)
Comment (anatolydyachenko) [reply DogServiceExample/pom.xml]: It's used in DogTest for assertion and @Test annotation... Substituted to:
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.4</version>
      <scope>test</scope>
    </dependency>

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 3 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: changed. incrementDogId() moved to DogService.

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 6 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: changed

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 9 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: Is it mandatory if there is only 1 statement after if part? Or it's some kind of patterns.

File: DogServiceExample/src/main/java/com/common/rest/DogService.java (hunk 13 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/rest/DogService.java]: Changed

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 4 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: Done

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 5 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: Done

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 7 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: Done

File: DogServiceExample/src/main/java/com/common/app/Dog.java (hunk 10 above)
Comment (anatolydyachenko) [reply DogServiceExample/src/main/java/com/common/app/Dog.java]: Done

**Pull request merged**